homeassistant/components/mysensors/cover.py

```diff
@@ -43,13 +53,44 @@ async def async_discover(discovery_info):
 class MySensorsCover(mysensors.device.MySensorsEntity, CoverEntity):
     """Representation of the value of a MySensors Cover child node."""
 
-    @property
-    def is_closed(self):
-        """Return True if cover is closed."""
+    def get_cover_state(self):
+        """Return a CoverState enum representing the state of the cover."""
         set_req = self.gateway.const.SetReq
+        v_up = self._values.get(set_req.V_UP) == STATE_ON
+        v_down = self._values.get(set_req.V_DOWN) == STATE_ON
+        v_stop = self._values.get(set_req.V_STOP) == STATE_ON
+
+        # If a V_DIMMER or V_PERCENTAGE is available, that is the amount
+        # the cover is open. Otherwise, use 0 or 100 based on the V_LIGHT
+        # or V_STATUS.
+        amount = 100
         if set_req.V_DIMMER in self._values:
-            return self._values.get(set_req.V_DIMMER) == 0
-        return self._values.get(set_req.V_LIGHT) == STATE_OFF
+            amount = self._values.get(set_req.V_DIMMER)
+        else:
+            amount = 100 if self._values.get(set_req.V_LIGHT) == STATE_ON else 0
+
+        if v_up and (not v_down) and (not v_stop):
+            return CoverState.OPENING
+        if (not v_up) and v_down and (not v_stop):
+            return CoverState.CLOSING
+        if (not v_up) and (not v_down) and v_stop and amount == 0:
```

Comment:
Looks like we can drop the parentheses:
https://docs.python.org/3/reference/expressions.html#operator-precedence
```suggestion
        if v_up and not v_down and not v_stop:
            return CoverState.OPENING
        if not v_up and v_down and not v_stop:
            return CoverState.CLOSING
        if not v_up and not v_down and v_stop and amount == 0:
```

Reply:
Why not. I'll add to the next batch.